backend/api/activity/index.js

```diff
@@ -347,6 +355,42 @@ activity.route('/search')
         error: 'Error retrieving from database',
       });
     }
+    
+    // Post-search culling (these checks require more complex calculations)
+    // Distance checks
+    if (maxDistance > 0) {
+        for (let i = 0; i < getobjs.length; i ++) {
+            //Figure out the zip codes we're comparing
+            let zip1 = getobjs[i].zipCode;
+            let zip2 = req.body.zipCode;
+            let dist = 0;
+            
+            try {
+                //Find these zip codes in the database
+                let zip1Obj = await ZipCode.findOne({ zipCode:zip1 });
+                let zip2Obj = await ZipCode.findOne({ zipCode:zip2 });
+                
+                //Find their latitudes and longitudes
+                let lat1 = zip1Obj.Latitude;
+                let lon1 = zip1Obj.Longitude;
+                let lat2 = zip2Obj.Latitude;
+                let lon2 = zip2Obj.Longitude;
+                
+                //Calculate their distance from each other
+                //TODO: Convert all degrees to radians in the dataset to speed up these calculations
+                dist = 3958.8 * Math.acos(Math.sin(lat1/57.2958) * Math.sin(lat2/57.2958) + Math.cos(lat1/57.2958) * Math.cos(lat2/57.2958) *  Math.cos(lon2/57.2958 - lon1/57.2958))
+            } catch (error) {
+                //If a zip code is invalid, consider them to be an infinite distance from each other
+                dist = 999999999;
+            }
+            
+            //If the activity is too far away, remove it from results
+            if (dist > maxDistance) {
+                getobjs.splice(i,1);
+                i --;
+            }
+        }
```

Comment:
Just a heads up, JavaScript has some really nice quality of life methods on basic types like arrays. One of them if [`.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) which accepts a callback and you can do all of the checking inside the callback. It'll create a new array with only the objects returned from the function callback so no need to use splice and such.